From f070e9627b78a2f17dbc586e8c6f09e919acf9df Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Fri, 31 Mar 2017 19:02:11 +0100
Subject: [PATCH] Do not save skin settings when Kodi is exiting. We instead
 save skin settings when changes are made. This change was added upstream to
 prevent hangs from causing loss of settings; but our sigterm handler already
 handles this as expected.

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/Application.cpp | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/xbmc/Application.cpp b/xbmc/Application.cpp
index b8ff91b..9b0417e 100644
--- a/xbmc/Application.cpp
+++ b/xbmc/Application.cpp
@@ -2904,13 +2904,6 @@ void CApplication::Stop(int exitCode)
     else
       CLog::Log(LOGNOTICE, "Not saving settings (settings.xml is not present)");
 
-    // kodi may crash or deadlock during exit (shutdown / reboot) due to
-    // either a bug in core or misbehaving addons. so try saving
-    // skin settings early
-    CLog::Log(LOGNOTICE, "Saving skin settings");
-    if (g_SkinInfo != nullptr)
-      g_SkinInfo->SaveSettings();
-
     m_bStop = true;
     m_AppFocused = false;
     m_ExitCode = exitCode;
-- 
2.7.4

